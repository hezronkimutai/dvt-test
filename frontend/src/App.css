.App {
  width: 80%;
  margin: auto;
}

.card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  margin: 20px auto;
}
.artist-card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  margin: 20px auto;
}

.card {
  width: 20%;
  max-width: 320px;
  min-width: 250px;
  margin: 10px 5px;
  border: none;
  outline: none;
  background-color: transparent;
  text-align: left;
  padding: 0;
  box-shadow: 1px 2px 5px 0px rgba(0, 0, 0, 0.329);
  -webkit-box-shadow: 1px 2px 5px 0px rgba(0, 0, 0, 0.329);
  -moz-box-shadow: 1px 2px 5px 0px rgba(0, 0, 0, 0.329);
  border-radius: 3px;
  cursor: pointer;
}
.card-text {
  padding: 2px 5px;
}
img {
  width: 100%;
}
.nav-bar {
  text-align: center;
  margin: 10px auto;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}
.search-box {
  width: min-content;
  margin: auto 5px;
  display: flex;
  flex-direction: row;
}
.search-box-button {
  margin: auto 3px;
  border: none;
  box-shadow: 1px 2px 5px 0px rgba(0, 0, 0, 0.329);
  -webkit-box-shadow: 1px 2px 5px 0px rgba(0, 0, 0, 0.329);
  -moz-box-shadow: 1px 2px 5px 0px rgba(0, 0, 0, 0.329);
  padding: 5px;
  cursor: pointer;
}
.search-box-input {
  border: none;
  box-shadow: 1px 2px 5px 0px rgba(0, 0, 0, 0.329);
  -webkit-box-shadow: 1px 2px 5px 0px rgba(0, 0, 0, 0.329);
  -moz-box-shadow: 1px 2px 5px 0px rgba(0, 0, 0, 0.329);
  border-radius: 3px;
  padding: 5px;
}

.albums {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  margin: 20px auto;
  position: relative;
}

.artist-details {
  background-color: black;
  opacity: 0.8;
  color: #fff;
  margin: auto;
  margin-top: -190px;
  padding: 20px;
  width: max-content;
  border-radius: 5px;
}
.potrait {
  width: 40%;
}
.potrait-img {
  max-height: 500px;
  object-fit: cover;
}
.tracks {
  width: 40%;
  text-align: left;
}
.back-btn {
  text-align: left !important;
}
.title {
  font-size: 40px;
  margin-top: 40px;
  width: 100%;
}
.logo {
  cursor: pointer;
}
.track {
  margin: 15px auto;
  padding: 5px;
  box-shadow: 1px 2px 5px 0px rgba(0, 0, 0, 0.329);
  -webkit-box-shadow: 1px 2px 5px 0px rgba(0, 0, 0, 0.329);
  -moz-box-shadow: 1px 2px 5px 0px rgba(0, 0, 0, 0.329);
  border-radius: 3px;
}
.album-title {
  padding: 5px 10px;
}
@media screen and (max-width: 700px) {
  .card {
    margin: 10px auto;
  }
  .search-box {
    margin: auto;
  }
  .potrait {
    width: 300px;
    margin: auto;
  }
  .tracks {
    width: 300px;
    margin: auto;
  }
  .back-btn {
    text-align: left !important;
  }
  .nav-bar {
    flex-direction: column;
  }
  .title {
    text-align: center;
  }
}
