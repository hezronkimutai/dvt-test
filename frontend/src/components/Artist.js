import { useLocation } from "react-router-dom";

function Artist() {
  const location = useLocation();

  const { state } = location;

  const tracks = (state && state.tracks && state.tracks.slice(0, 5)) || [];

  return (
    <>
      <div className="artist-card-container">
        <div className="potrait">
          <img className="potrait-img" src={state && state.picture_xl} />
          <section className="artist-details">
            <h3>{state && state.name}</h3>
            <p><b>{state && state.nb_fan}</b> Fans</p>
            <p><b>{state && state.nb_album}</b> Albums</p>
          </section>
        </div>
        <div className="tracks">
          <h1 className="title">Top Five Tracks</h1>
          {tracks.map(track => (
            <div className="track">
              <span>{track.title}</span>
            </div>)
          )}
        </div>
      </div>
      <div className="albums">
        <h1 className="title">Albums</h1>
        {tracks.map(track => (
          <div className="card">
            <h3 className="album-title">{track.album.title}</h3>
            <img src={track.album.cover_xl} />
          </div>)
        )}</div>
    </>
  );
}

export default Artist;
