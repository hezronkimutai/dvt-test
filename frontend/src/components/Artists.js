import React from 'react';
import { useNavigate } from 'react-router-dom';
import { AppContext } from '../utils';

function Artists() {
  const { data } = React.useContext(AppContext);
  const navigate = useNavigate();

  const handleGetDetails = (e, artist) => {
    e.preventDefault();
    navigate(`/${artist.id}`, { state: artist });
  }

  return (
    <>
      <div className="card-container">
        {data && data.artists
          ? data.artists.map(artist => (
            <button onClick={(e) => handleGetDetails(e, artist)} className="card">
              <img src={artist.picture_xl} />
              <p className="card-text"><b>Name :</b>{artist.name}</p>
              <p className="card-text"><b>Fans :</b> {artist.nb_fan}</p>
              <p className="card-text"><b>Albums :</b> {artist.nb_album}</p>
            </button>))
          : <h2>Loading....</h2>}
      </div>
    </>
  );
}


export default Artists;
