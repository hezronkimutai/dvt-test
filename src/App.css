.App {
  text-align: center;
  width: 80%;
  margin: auto;
}

.card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  margin: 20px auto;
}

.card {
  width: 20%;
  max-width: 300px;
  margin: auto;
}
img {
  width: 100%;
}
