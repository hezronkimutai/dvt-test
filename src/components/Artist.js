import { useLocation, useNavigate } from "react-router-dom";

function Artist() {
  const navigate = useNavigate();
  const location = useLocation();

  const { state } = location;

  const handleGoBack = () => {
    navigate("/");
  }

  return (
    <>
      <button onClick={handleGoBack}>Back</button>
      <div className="card-container">
        <div><img style={{ maxWidth: 320 }} src={state.picture_xl} /></div>
        <ol>{state.tracks.map(track => (
          <li>
            <span>{track.title}</span>
            <span>{track.rank}</span>
          </li>)
        )}</ol>
        <h2>Albums</h2>
        <div>{state.tracks.map(track => (
          <div>
            <span>{track.album.title}</span>
            <img style={{ maxWidth: 320 }} src={track.album.cover_xl} />
          </div>)
        )}</div>
      </div>
    </>
  );
}

export default Artist;
