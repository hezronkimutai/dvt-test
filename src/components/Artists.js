import axios from 'axios';
import { useEffect, useState } from 'react';
import React from 'react';
import { useNavigate } from 'react-router-dom';

const url = 'https://shrouded-hamlet-74185.herokuapp.com/';

function Artists() {
  const navigate = useNavigate();

  const [searchTerm, setSearchTerm] = useState('eminem');
  const [data, setData] = useState({});

  useEffect(() => {
    fetchData(searchTerm);
  }, [])

  const handleChange = (e) => {
    e.preventDefault();
    const { value } = e.target;
    setSearchTerm(value);
  }

  const fetchData = (search) => {
    axios.post(url, { searchTerm: search }).then(res => {
      setData(res.data);
    });
  }
  const handleSearch = () => {
    fetchData(searchTerm);
  }

  const handleGetDetails = (e, artist) => {
    e.preventDefault();
    navigate(`/${artist.id}`, { state: artist });
  }

  return (
    <>
      <div className="search-box">
        <input className="search-box-input" type='text' value={searchTerm} onChange={handleChange} />
        <button className="search-box-button" onClick={handleSearch}>Search</button>
      </div>
      <div className="card-container">
        {data.artists && data.artists.map(artist => (
          <button onClick={(e) => handleGetDetails(e, artist)} className="card">
            <img src={artist.picture_xl} />
            <p className="card-text"><b>Name :</b>{artist.name}</p>
            <p className="card-text"><b>Fans :</b> {artist.nb_fan}</p>
            <p className="card-text"><b>Albums :</b> {artist.nb_album}</p>
          </button>))}
      </div>
    </>
  );
}


export default Artists;
