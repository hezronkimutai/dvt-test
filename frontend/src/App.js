import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import React from 'react';
import Artist from './components/Artist';
import Artists from './components/Artists';
import NavBar from './components/NavBar';
import { useEffect, useState } from 'react';
import { AppContext, fetchData } from './utils';

import './App.css';

const App = (props) => {
  const [searchTerm, setSearchTerm] = useState('eminem');
  const [data, setData] = useState({});

  const initialState = { searchTerm, setSearchTerm, data, setData };

  useEffect(() => {
    fetchData(searchTerm, setData);
  }, []);

  return (
    <AppContext.Provider value={initialState}>
      <div className="App">
        <Router >
          <NavBar />
          <Routes>
            <Route path="/" exact element={<Artists data={data} {...props} />} />
            <Route exact path="/:artistId" element={<Artist {...props} />} />
          </Routes>
        </Router>
      </div>
    </AppContext.Provider>
  )
}

export default App;
