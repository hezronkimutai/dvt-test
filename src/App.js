import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import React from 'react';
import Artist from './components/Artist';
import Artists from './components/Artists'

import './App.css';



const App = () => {
  return (

    <Router>
      <Switch>

        <Route path="/" component={Artists} />
        {/* <Route exact path="/:artistId" component={Artist} />
        <Route exact path="/" component={NavBar} />
        <Route path="/page-1" component={PageOne} />
        <Route path="/page-2" component={PageTwo} /> */}
      </Switch>
    </Router>
  )
}

export default App;
