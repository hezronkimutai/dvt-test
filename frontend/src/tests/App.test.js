import { render } from '@testing-library/react';
import App from '../App';
import Artist from '../components/Artist';
import Artists from '../components/Artists';
import { BrowserRouter as Router } from 'react-router-dom'

test('matches the snapshot', () => {
  const component = render(<App />);
  expect(component).toMatchSnapshot();
});

test('matches the snapshot', () => {
  const component = render(
    <Router >
      <Artist />
    </Router>);
  expect(component).toMatchSnapshot();
});

test('matches the snapshot', () => {
  const component = render(
    <Router >
      <Artists />
    </Router>);
  expect(component).toMatchSnapshot();
});