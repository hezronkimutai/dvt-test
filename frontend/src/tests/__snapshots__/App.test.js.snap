// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <nav
          class="nav-bar"
        >
          <h1
            class="logo"
          >
            Deezer
          </h1>
          <div
            class="search-box"
          >
            <input
              class="search-box-input"
              type="text"
              value="eminem"
            />
            <button
              class="search-box-button"
            >
              Search
            </button>
          </div>
        </nav>
        <div
          class="card-container"
        >
          <h2>
            Loading....
          </h2>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <nav
        class="nav-bar"
      >
        <h1
          class="logo"
        >
          Deezer
        </h1>
        <div
          class="search-box"
        >
          <input
            class="search-box-input"
            type="text"
            value="eminem"
          />
          <button
            class="search-box-button"
          >
            Search
          </button>
        </div>
      </nav>
      <div
        class="card-container"
      >
        <h2>
          Loading....
        </h2>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`matches the snapshot 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="artist-card-container"
      >
        <div
          class="potrait"
        >
          <img
            class="potrait-img"
          />
          <section
            class="artist-details"
          >
            <h3 />
            <p>
              <b />
               Fans
            </p>
            <p>
              <b />
               Albums
            </p>
          </section>
        </div>
        <div
          class="tracks"
        >
          <h1
            class="title"
          >
            Top Five Tracks
          </h1>
        </div>
      </div>
      <div
        class="albums"
      >
        <h1
          class="title"
        >
          Albums
        </h1>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="artist-card-container"
    >
      <div
        class="potrait"
      >
        <img
          class="potrait-img"
        />
        <section
          class="artist-details"
        >
          <h3 />
          <p>
            <b />
             Fans
          </p>
          <p>
            <b />
             Albums
          </p>
        </section>
      </div>
      <div
        class="tracks"
      >
        <h1
          class="title"
        >
          Top Five Tracks
        </h1>
      </div>
    </div>
    <div
      class="albums"
    >
      <h1
        class="title"
      >
        Albums
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`matches the snapshot 3`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-container"
      >
        <h2>
          Loading....
        </h2>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-container"
    >
      <h2>
        Loading....
      </h2>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
